[Actionable Observability] update alerts table rule details link to point to o11y rule detail page

mgiota opened the pull request:
Fixes https://github.com/elastic/kibana/issues/132389

mgiota committed:
update alerts table rule details link to point to o11y rule detail page

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

mgiota committed:
o11y alert flyout should also link to o11y rule details page

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

mgiota committed:
add data-test-subj to rule details page title and add move path definition

mgiota commented on x-pack/test/observability_functional/apps/observability/alerts/index.ts:
```diff
@@ -223,7 +223,9 @@ export default ({ getService }: FtrProviderContext) => {
           await actionsButton.click();
           await observability.alerts.common.viewRuleDetailsButtonClick();
-          expect(await find.existsByCssSelector('[title="Rules and Connectors"]')).to.eql(true);
+          expect(await find.existsByCssSelector('[data-test-subj="ruleDetailPageTitle"]')).to.eql(
```
@fkanout Currently there's a problem with current test which keeps failing because rule is undefined and rule details page can not be loaded. Here's where rule is undefined https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/rule_details/index.tsx#L201. Stack Management Rule details page is getting the rule as param https://github.com/elastic/kibana/blob/main/x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_details.tsx#L76 and I guess that's why the old detail page could load in the tests without issues. This refactoring is out of scope of this issue. How do you suggest we proceed with this failing test?  

<img width="1712" alt="Screenshot 2022-05-19 at 08 43 03" src="https://user-images.githubusercontent.com/2852703/169228805-3b6b2481-985e-4cf0-aec0-3ce8cb697ae2.png">

mgiota replied:
I just pushed a fix that checks if Observability is in the breadcrumb https://github.com/elastic/kibana/pull/132479/commits/1f94a1f2bc0e9b93868a3b52b9e3c20de2984cc9#diff-5a75bdfd91813e385e6c322eed601a2fd55d34f4a56b7669b0bb193014df1648R226

fkanout replied:
@mgiota, I think this ok for the purpose of this test and in the scope of this PR.

mgiota committed:
fix failing tests by checking existance of Observability in breadcrumb

fkanout commented on x-pack/plugins/observability/public/config/paths.ts:
```diff
@@ -8,6 +8,7 @@
 export const paths = {
   observability: {
     alerts: '/app/observability/alerts',
+    ruleDetails: (ruleId: string) => `/app/observability/alerts/rules/${encodeURI(ruleId)}`,
```
What do you think of using this var for the URL https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/rule_details/config.ts#L22?

mgiota replied:
Yep great suggestion! I'll change it

fkanout commented on x-pack/plugins/observability/public/pages/alerts/components/alerts_flyout/alerts_flyout.tsx:
```diff
@@ -77,8 +77,8 @@ export function AlertsFlyout({
-  const linkToRule = ruleId && prepend ? prepend(paths.management.ruleDetails(ruleId)) : null;
-
+  const linkToRule =
+    ruleId && prepend ? prepend(`/app/observability/alerts/rules/${ruleId}`) : null;
```
Same here https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/rule_details/config.ts#L22 ?

mgiota replied:
I'll use `prepend(paths.observability.ruleDetails(ruleId))`, so that I need to do the path change only in the paths file.

Ideally we should move these variables in a common place in observability plugin, because they are used by both rules and alerts pages. I will create another ticket to replace all hardcoded values of /app/observability/alerts/ and /app/observability/alerts/rules and this should be covered there.

fkanout reviewed (approved):
LGTM. Tested, and it worked as expected. Thank you for spotting the need for these changes!
One suggestion about using the URL var instead of the hardcoded one

mgiota committed:
use alerts and rules link from the paths file

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

mgiota committed:
update link in alert flyout to use paths

mgiota committed:
update rule details link in the rules page

mgiota commented:
I moved the variables in the common [paths](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/config/paths.ts) file and I refactored following places to make use of it:
- rule details link in alerts table
- rule details link in alerts flyout
- rules details link in rules page
- rule details link and alerts link in rule details page
